pinot-core/src/main/java/org/apache/pinot/core/segment/memory/PinotByteBuffer.java

```diff
@@ -327,8 +327,11 @@ public void flush() {
 
   @Override
   protected void release() {
-    if (((DirectBuffer) _buffer).cleaner() != null) {
-      ((DirectBuffer) _buffer).cleaner().clean();
+    if (CleanerUtil.UNMAP_SUPPORTED) {
+      try {
+        CleanerUtil.getCleaner().freeBuffer(_buffer);
+      } catch (IOException e) {
```

Comment:
Will this throw exception in normal case? We should not swallow exception

Reply:
will log the error

Reply:
We should throw the IOException out (the interface expects IOException to be thrown)

Reply:
will do